scripts/outer/kubectl-kubestellar-prep_for_syncer

```diff
@@ -111,89 +111,67 @@ fi
 set -e
 
 imw_space_config="${PWD}/temp-space-config/spaceprovider-default-${imw}"
-kubectl-kubestellar-get-config-for-space --space-name ${imw} --provider-name default --sm-core-config $SM_CONFIG --sm-context $SM_CONTEXT --space-config-file $imw_space_config
+kubectl-kubestellar-get-config-for-space --space-name ${imw} --provider-name default --sm-core-config $SM_CONFIG --sm-context $SM_CONTEXT --output $imw_space_config
 mbwsname=$(KUBECONFIG=$imw_space_config kubectl "${kubectl_flags[@]}" get synctargets.edge.kubestellar.io "$stname" -o jsonpath="{.metadata.annotations['kcp\.io/cluster']}-mb-{.metadata.uid}")
 
-original_ws=$(kubectl ws "${kubectl_flags[@]}" . --short)
-
-if [ "$imw" == "." ]
-then imw="$original_ws"
-elif ! [[ "$imw" =~ [a-z0-9].* ]]; then
+if ! [[ "$imw" =~ [a-z0-9].* ]]; then
     echo "$0: imw ${espw@Q} is not valid" >&2
     exit 1
 fi
 
-if [ "$espw" == "." ]
-then espw="$original_ws"
-elif ! [[ "$espw" =~ [a-z0-9].* ]]; then
+if ! [[ "$espw" =~ [a-z0-9].* ]]; then
     echo "$0: espw ${espw@Q} is not valid" >&2
     exit 1
 fi
 
-if [ "$silent" == "true" ]; then
-    trap 'kubectl ws "${kubectl_flags[@]}" "$original_ws" > /dev/null' EXIT
-else
-    trap 'kubectl ws "${kubectl_flags[@]}" "$original_ws"' EXIT
-fi
-
-if [ "$silent" == "true" ]; then
-    kubectl ws "${kubectl_flags[@]}" "${rootws}:${imw}" > /dev/null
-else
-    kubectl ws "${kubectl_flags[@]}" "${rootws}:${imw}"
-fi
-kubectl ws "${kubectl_flags[@]}" "${rootws}:${imw}"
-spaceID=$(kubectl "${kubectl_flags[@]}" get synctargets.edge.kubestellar.io "$stname" -o jsonpath="{.metadata.annotations['kcp\.io/cluster']}")
+espw_space_config="${PWD}/temp-space-config/spaceprovider-default-${espw}"
+kubectl-kubestellar-get-config-for-space --space-name ${espw} --provider-name default --sm-core-config $SM_CONFIG --sm-context $SM_CONTEXT --output $espw_space_config
 
-kubectl ws "${kubectl_flags[@]}" "${rootws}:${espw}"
-if ! cluster_ns_name="$(kubectl "${kubectl_flags[@]}" -n kubestellar get configmap kbmap-"$spaceID" -ojsonpath="{.metadata.labels['kubestellar\.io\/kube-bind-id']}")"; then
-	echo "error reading kube-bind cluster namespace from ConfigMap kbmap-$spaceID" >&2
+if ! cluster_ns_name="$(KUBECONFIG=$espw_space_config kubectl "${kubectl_flags[@]}" -n kubestellar get configmap kbmap-"$imw" -ojsonpath="{.metadata.labels['kubestellar\.io\/kube-bind-id']}")"; then
+	echo "error reading kube-bind cluster namespace from ConfigMap kbmap-$imw" >&2
 	exit 1
 fi
+
 prefixed_stname="$cluster_ns_name"-"$stname"
-stUID=$(kubectl "${kubectl_flags[@]}" get synctargets.edge.kubestellar.io $prefixed_stname -o jsonpath="{.metadata.uid}")
-mbwsname="$spaceID"-mb-"$stUID"
+stUID=$(KUBECONFIG=$espw_space_config kubectl "${kubectl_flags[@]}" get synctargets.edge.kubestellar.io $prefixed_stname -o jsonpath="{.metadata.uid}")
```

Comment:
The `"${kubectl_flags[@]}"` here is left-over confusion. You do not need it to get kubectl to access the KCS, right?